fury/actors/tensor.py

```diff
@@ -211,3 +211,288 @@ def tensor_ellipsoid(centers, axes, lengths, colors, scales, opacity):
+
+    # Number of vertices that make up the box
+    n_verts = 8
+
+    big_centers = np.repeat(centers, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_centers, 'center')
+
+    big_scales = np.repeat(scales, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_scales, 'scale')
+
+    evec1 = np.array([item[0] for item in axes])
+    evec2 = np.array([item[1] for item in axes])
+    evec3 = np.array([item[2] for item in axes])
+
+    big_vectors_1 = np.repeat(evec1, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_vectors_1, 'evec1')
+    big_vectors_2 = np.repeat(evec2, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_vectors_2, 'evec2')
+    big_vectors_3 = np.repeat(evec3, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_vectors_3, 'evec3')
+
+    big_angles = np.repeat(np.array(angles, dtype=float), n_verts, axis=0)
+    attribute_to_actor(box_actor, big_angles, 'angle')
+
+    # Start of shader implementation
+
+    vs_dec = \
+        """
+        in vec3 center;
+        in float scale;
+        in vec3 evec1;
+        in vec3 evec2;
+        in vec3 evec3;
+        in float angle;
+
+        out vec4 vertexMCVSOutput;
+        out vec3 centerMCVSOutput;
+        out float scaleVSOutput;
+        out mat3 rotationMatrix;
+        out float angleVSOutput;
+        """
+
+    # Variables assignment
+    v_assign = \
+        """
+        vertexMCVSOutput = vertexMC;
+        centerMCVSOutput = center;
+        scaleVSOutput = scale;
+        angleVSOutput = angle;
+        """
+
+    # Rotation matrix
+    rot_matrix = \
+        """
+        mat3 R = mat3(normalize(evec1), normalize(evec2), normalize(evec3));
+        float a = radians(90);
+        mat3 rot = mat3(cos(a),-sin(a), 0,
+                        sin(a), cos(a), 0, 
+                            0 ,      0, 1);
+        rotationMatrix = transpose(R) * rot;
+        """
+
+    vs_impl = compose_shader([v_assign, rot_matrix])
+
+    shader_to_actor(box_actor, 'vertex', decl_code=vs_dec,
+                    impl_code=vs_impl)
+
+    fs_vars_dec = \
+        """
+        in vec4 vertexMCVSOutput;
+        in vec3 centerMCVSOutput;
+        in float scaleVSOutput;
+        in mat3 rotationMatrix;
+        in float angleVSOutput;
+
+        uniform mat4 MCVCMatrix;
+        """
+
+    # Importing the cone SDF
+    sd_cone = import_fury_shader(os.path.join('sdf', 'sd_cone.frag'))
+
+    # Importing the union operation SDF
+    sd_union = import_fury_shader(os.path.join('sdf', 'sd_union.frag'))
+
+    # SDF definition
+    sdf_map = \
+        """
+        float map(in vec3 position)
+        {
+            vec3 p = (position - centerMCVSOutput)/scaleVSOutput
+                *rotationMatrix;
+            float angle = clamp(angleVSOutput, 0, 6.283);
+            vec2 a = vec2(sin(angle), cos(angle));
+            float h = .5 * a.y;
+            return opUnion(sdCone(p,a,h), sdCone(-p,a,h)) * scaleVSOutput;
+        }
+        """
```

Comment:
This look like a duplicate. I am surprised that you do not import this shader. Something to check or to improve.